lib/Conversion/CalyxToFSM/CalyxToFSM.cpp

```diff
@@ -279,6 +347,8 @@ void CalyxToFSMPass::runOnOperation() {
   auto ctrlOp = component.getControlOp();
   assert(ctrlOp.getBodyBlock()->getOperations().size() == 1 &&
          "Expected a single top-level operation in the schedule");
+  CompileInvoke compileInvoke(component, builder);
```

Comment:
@mortbopet does this seem like the right place to put the invoke compilation? In the native compiler, compiling invokes is a separate pass entirely so that we can perform optimizations that don't support `invoke` but do support other operations.

Reply:
This is just a lowering `Invoke` right? IMO, we should keep it here until we need a separate pass.